[OSSMC] View in tracing integration with the tracing UI

josunect opened the pull request:
### Describe the change

- Detects if the Distributed tracing is installed getting the plugin manifest
- When tracing action is sent, it uses the plugin configuration to create and redirect to the tracing plugin.
![image](https://github.com/user-attachments/assets/27a06427-0daa-42ae-85de-fc23aac2ec4a)

### Steps to test the PR

(Note: not really simple to test)

- For testing locally, uncomment distributedTracingPluginConfig setup (Locally, I was not sure how to enable other plugins, even if this is installed). 
- Copy Kiali PR code: https://github.com/kiali/kiali/pull/8265
- Install in OpenShift Kiali with Tempo: 
https://medium.com/kialiproject/installing-openshift-service-mesh-3-tp1-with-kiali-and-grafana-tempo-6b76881ceaef
One tenant name should be "default" to work with the defaults
- Install Distributed tracing UI plugin: 
https://docs.redhat.com/en/documentation/openshift_container_platform/4.14/html/cluster_observability_operator/observability-ui-plugins#coo-distributed-tracing-ui-plugin-using_distributed-tracing-ui-plugin
- Generate traces
- Go to Kiali UI > See traces from an application. Test that the links are working for "View in tracing" and "More span details" 
![image](https://github.com/user-attachments/assets/cc53736e-5702-4795-81d9-32909f340a77)
![image](https://github.com/user-attachments/assets/6f5c4f71-33d6-4b00-b97c-02c296270b29)
![image](https://github.com/user-attachments/assets/5e12f494-c651-44eb-a75b-3d29ba272234)

### Automation testing

If applicable, explain the case scencarios covered by **e2e** tests created for this PR.

### Issue reference

Ref. https://github.com/kiali/kiali/issues/7904

ferhoyos reviewed (commented):
Great job @josunect! Cross-plugin work was something we had been looking forward to achieving for a long time!

For now, I have tested the feature, and it looks great. Clicking the tracing link redirects to the Observability OpenShift plugin if it exists.

![image](https://github.com/user-attachments/assets/ffd78c71-8227-4253-94ec-578e2bd56e7d)

![image](https://github.com/user-attachments/assets/f863a673-27f4-44d7-a7e9-84542099e7ed)

The only issue I see is that you have to click the `Back` button twice to return to the previous OSSMC page. This is probably related to the `react-router` library. Anyway, I consider this a minor issue, and we can create a follow-up PR to address it.

Tomorrow, I will review the code before giving final approval.

ferhoyos commented on plugin/src/openshift/utils/KialiIntegration.tsx:
```diff
@@ -124,6 +156,27 @@ export const useInitKialiListeners = (): void => {
+            } else {
+              consoleUrl = `/observe/traces?namespace=${namespace}&name=${instance}&tenant=${tenant}&q=%7B%7D&limit=20`
+            }
+            setTimeout(() => navigate(consoleUrl), 0);
```
This line shouldn't be here, you are already calling to `navigate` in line 183. Maybe this is the reason why you have to click twice to the `Back` button to return to the OSSMC plugin.

ferhoyos commented on plugin/src/openshift/utils/KialiIntegration.tsx:
```diff
@@ -124,6 +156,27 @@ export const useInitKialiListeners = (): void => {
+            const namespace = pluginConfig.observability.namespace;
+            const instance = pluginConfig.observability.instance;
+            const trace = urlParams.get('trace');
+            const tenant = pluginConfig.observability.tenant ? pluginConfig.observability.tenant : "default"
```
Added Semicolon and simpler structure
```suggestion
            const tenant = pluginConfig.observability.tenant ?? "default";
            
```

ferhoyos commented on plugin/src/openshift/components/KialiController.tsx:
```diff
@@ -139,11 +158,51 @@ class KialiControllerComponent extends React.Component<KialiControllerProps> {
+        .then(response => (distributedTracingPluginConfig = (response)))
+        .catch(error => {
+          console.debug(`Error fetching Distributed Tracing plugin configuration. (Probably is not installed) ${error}`)
+          // Enable this for testing locally
```
Does it mean you have to uncomment this piece of code to test OSSMC locally in any case? or only when you want to test Distributed Tracing plugin locally? I don't like much to have such comment code here, maybe it is better to add it to the README doc. What do you think?

josunect replied:
Yes, I can move it to the README. This is just required for the Distributed Tracing integration testing.

ferhoyos reviewed (changes requested):
Thanks Josune for this PR! Overall, looks good, just a pair of comments to be addressed.

josunect commented:
@ferhoyos I have addressed the comments. 

Also, I've made a change to try to auto discover the url from the tracing internal url. I let the configuration in case is needed if the discovery url for some reason doesn't work.

ferhoyos reviewed (approved):
With the latest changes I do not see the `Back` button issue. The code looks fine for me. Thanks @josunect for bringing this long-awaited feature to OSSMC!

ferhoyos commented on plugin/src/openshift/utils/__tests__/KialiIntegration.test.ts:
```diff
@@ -0,0 +1,45 @@
+    expect(parsed?.instance).toEqual('sample');
+  });
+
+  it('url for multi tenant url for should be correct', () => {
```
typo in the test description?

ferhoyos reviewed (approved):
LGTM
